tests/std/tests/P0896R4_views_iota/test.cpp

```diff
@@ -0,0 +1,294 @@
+        static_assert(noexcept(first <=> second)); // strengthened
+
+        {
+            I i = first;
+
+            assert(&++i == &i);
+            assert(i == second);
+            static_assert(noexcept(++i)); // strengthened
+
+            i = first;
+
+            assert(i++ == first);
+            assert(i == second);
+            static_assert(noexcept(i++)); // strengthened
+
+            assert(&--i == &i);
+            assert(i == first);
+            static_assert(noexcept(--i)); // strengthened
+
+            i = second;
+
+            assert(i-- == second);
+            assert(i == first);
+            static_assert(noexcept(i--)); // strengthened
+
+            assert(i + 1 == second);
+            static_assert(noexcept(i + 1)); // strengthened
+
+            assert(1 + i == second);
+            static_assert(noexcept(1 + i)); // strengthened
+
+            assert(&(i += 1) == &i);
+            assert(i == second);
+            static_assert(noexcept(i += 1)); // strengthened
+            assert(i - 1 == first);
+            static_assert(noexcept(i - 1)); // strengthened
+
+            assert(&(i -= 1) == &i);
+            assert(i == first);
+            static_assert(noexcept(i -= 1)); // strengthened
+
+            assert(second - first == 1);
+            static_assert(noexcept(second - first)); // strengthened
+
+            assert(first[1] == *second);
+            assert(second[-1] == *first);
+            static_assert(noexcept(first[1])); // strengthened
+        }
+
+        const same_as<ranges::sentinel_t<R>> auto last = rng.end();
+        assert(!(first == last));
+        static_assert(noexcept(first == last)); // strengthened
+        assert(first != last);
+        static_assert(noexcept(first != last)); // strengthened
+        assert(last - first == 8);
+        static_assert(noexcept(last - first)); // strengthened
+    }
+
+    {
+        // Validate unbounded (upper bound is unreachable) iota_view
+        using R = ranges::iota_view<T>;
+        static_assert(same_as<ranges::range_value_t<R>, T>);
+        static_assert(same_as<ranges::range_reference_t<R>, T>);
+
+        static_assert(ranges::random_access_range<R>);
+        static_assert(!ranges::contiguous_range<R>);
+        static_assert(!ranges::common_range<R>);
+        static_assert(same_as<ranges::sentinel_t<ranges::iota_view<T>>, unreachable_sentinel_t>);
+
+        // Bounded and unbounded iota_view have the same iterator type, however:
+        static_assert(same_as<ranges::iterator_t<ranges::iota_view<T, T>>, ranges::iterator_t<ranges::iota_view<T>>>);
+
+        if constexpr (sizeof(T) < sizeof(int)) {
+            static_assert(same_as<ranges::range_difference_t<R>, int>);
+        } else {
+            static_assert(same_as<ranges::range_difference_t<R>, long long>);
+        }
+
+        {
+            const same_as<R> auto rng = views::iota(low);
+
+            auto i = low;
+            for (auto const& e : rng) {

```

Comment:
```suggestion
            for (const auto& e : rng) {
```

Reply:
I thought you said nitpicks, Stephan.

Reply:
`const`-west is `const`-worst; east-`const` is beast-`const`!